### ISSUE TITLE
Data Source azurerm_kubernetes_cluster nil pointer

### ISSUE THREAD

**Issue by jschwanz** [hashicorp/terraform-provider-azurerm#21840]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.4.6

### AzureRM Provider Version

3.56.0

### Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster

### Terraform Configuration Files

```hcl
data "azurerm_kubernetes_cluster" "my_aks" {
  name                = "my-aks-cluster"
  resource_group_name = "my-resource-group"
}
```

### Debug Output/Panic Output

```shell
The plugin.(*GRPCProvider).ReadDataSource request was cancelled.

Error: Plugin did not respond

  with data.azurerm_kubernetes_cluster.my_aks,
  on main.tf line 47, in data "azurerm_kubernetes_cluster" "my_aks":
  47: data "azurerm_kubernetes_cluster" "my_aks" {

The plugin encountered an error, and failed to respond to the
plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more
details.

Stack trace from the terraform-provider-azurerm_v3.56.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4fb7741]

goroutine 87 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/containers.flattenKubernetesClusterDataSourceStorageProfile(0xc001da3e00)
	github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/kubernetes_cluster_data_source.go:900 +0x81
github.com/hashicorp/terraform-provider-azurerm/internal/services/containers.dataSourceKubernetesClusterRead(0x0?, {0x6207700?, 0xc002554c00?})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/kubernetes_cluster_data_source.go:796 +0x17f9
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x768e0a0?, {0x768e0a0?, 0xc001235830?}, 0xd?, {0x6207700?, 0xc002554c00?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:712 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc000c77ce0, {0x768e0a0, 0xc001235830}, 0xc0020a2500, {0x6207700, 0xc002554c00})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:943 +0x145
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc0010a0768, {0x768e0a0?, 0xc0012352f0?}, 0xc0020fe200)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:1195 +0x38f
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0xc0010c3540, {0x768e0a0?, 0xc001234480?}, 0xc0004b8870)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:658 +0x403
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0x69b2860?, 0xc0010c3540}, {0x768e0a0, 0xc001234480}, 0xc00293c070, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:421 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00015c1e0, {0x769dd60, 0xc0019f01a0}, 0xc001be4240, 0xc001b185a0, 0xbbbbad0, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc00015c1e0, {0x769dd60, 0xc0019f01a0}, 0xc001be4240, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.51.0/server.go:963 +0x28a

Error: The terraform-provider-azurerm_v3.56.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Actual Behaviour

I get this same error regardless of region, and it occurs as far back as azurerm 3.46.0

### Steps to Reproduce

terraform plan

**Comment by katbyte** [comment]
Thanks @jschwanz for letting us know @jschwanz - reopening

**Comment by lonegunmanb** [comment]
> Thanks @jschwanz for letting us know @jschwanz - reopening
> 
> @lonegunmanb - can you take another look?

Sure thing @katbyte @, I'll

**Comment by lonegunmanb** [comment]
Apology for the inconveniency @jschwanz, #21850 failed to solve your issue because I missed some nil checks, and since I don't have the configuration to reproduce this panic on my side I didn't do the test to verify the fix. I think #21867 could solve this nil panic once for all.

**Comment by jschwanz** [comment]
Confirmed this is now fixed and working as expected in 3.58.0.
